.github/workflows/main.yml

```diff
@@ -0,0 +1,44 @@
+name: Create and publish a Docker image
+
+on:
+  push:
+    branches:
+      - '*'
+    tags:
+      - '*'
+  pull_request:
+    branches:
+      - 'main'
+
+concurrency: ci-${{ github.ref }}
+
+jobs:
+  tests:
```

Comment:
i still don't see a docker build step, e.g. `docker build -t [image_name] .`, in the tests job?  It should get triggered on a PR, a push to any branch and a tag.

Reply:
I think it always builds and only pushes when the 'push' boolean is true. That would achieve the same as a build step in the test step would it not?

Reply:
@zaro0508 It's the `docker/build-push-action@v3.2.0` in `docker_build.yml` that builds and pushes the image.  Does that make sense?

Reply:
> I think it always builds and only pushes when the 'push' boolean is true. That would achieve the same as a build step in the test step would it not?

it's not the same because docker build will not run as part of the tests job.  it would run in the docker-build-jupyter or docker-build-rstudio jobs which may potentially push as well which is confusing.  to simpfly things we should also add docker build to the tests job.